buildSrc/src/main/kotlin/io/spine/internal/gradle/github/pages/GitHubPagesBranch.kt

```diff
@@ -0,0 +1,188 @@
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.github.pages
+
+import io.spine.internal.gradle.Cli
+import io.spine.internal.gradle.RepoSlug
+import java.io.File
+import org.gradle.api.GradleException
+
+/**
+ * Represents the branch used for publishing documentation.
+ *
+ * The repository's GitHub URL is derived from the `REPO_SLUG` environment variable.
+ * The branch dedicated to publishing documentation is automatically checked out in
+ * this repository. Also, the username and email of the git committer are
+ * automatically configured. The username is set to "UpdateGitHubPages Plugin", and
+ * the email is derived from the `FORMAL_GIT_HUB_PAGES_AUTHOR` environment variable.
+ *
+ * This class provides functionality to commit changes in the specific directory and
+ * push them to the remote repository.
+ */
+internal class GitHubPagesBranch {
+
+    /** The branch dedicated to publishing documentation. */
+    private val targetBranch = "gh-pages"
+    val repoFolder: File
+
+    internal constructor(rootFolder: File, repoFolder: File) {
+        this.repoFolder = repoFolder
+
+        SshKey(rootFolder).register()
+        clone(rootFolder)
+        checkout()
+        configureCommitter()
+    }
+
+    /**
+     * Clones the Git repository with the GitHub URL retrieved from the `REPO_SLUG`
+     * environment variable.
+     */
+    private fun clone(rootFolder: File) {
+        val gitHost = RepoSlug.fromVar().gitHost()
+        Cli(rootFolder).execute("git", "clone", gitHost, repoFolder.absolutePath)
+    }
+
+    /**
+     * Configures Git to publish the changes under "UpdateGitHubPages Plugin" Git
+     * username and email stored in the `FORMAL_GIT_HUB_PAGES_AUTHOR` environment
+     * variable.
+     */
+    private fun configureCommitter() {
+        pagesExecute("git", "config", "user.name", "\"UpdateGitHubPages Plugin\"")
+        val authorEmail = AuthorEmail.fromVar().toString()
+        pagesExecute("git", "config", "user.email", authorEmail)
+    }
+
+    /** Executes a command in the [repoFolder]. */
+    private fun pagesExecute(vararg command: String): String = Cli(repoFolder).execute(*command)
+
+    private fun checkout() {
+        pagesExecute("git", "checkout", targetBranch)
+    }
+
+    /**
+     * Stages all changes in the provided directory and commits with the provided
+     * message.
+     */
+    fun commit(directory: String, message: String) {
```

Comment:
This API piece is weird. In a real world, committing isn't a functionality of a branch, for sure.

Also, I would think that if I clone the repository locally into some folder, and then I checkout some branch, I would then have no particular choice of some "provided directory". All the changes that I have made in the folder to which I originally cloned the repo, those are mine to commit.

Please review. Let's discuss vocally if needed.